security/s2a/internal/crypter/expander.go

```diff
@@ -12,3 +14,24 @@ type hkdfExpander interface {
 	// in number of bytes, and returns the resulting expanded key.
 	expand(hash func() hash.Hash, secret, label []byte, length int) ([]byte, error)
 }
+
+// defaultHKDFExpander is the default HKDF expander which uses Go's crypto/hkdf
+// for HKDF expansion.
+type defaultHKDFExpander struct{}
+
+// newDefaultHKDFExpander creates an instance of the default HKDF expander.
+func newDefaultHKDFExpander() hkdfExpander {
+	return &defaultHKDFExpander{}
+}
+
+func (*defaultHKDFExpander) expand(hash func() hash.Hash, secret, label []byte, length int) ([]byte, error) {
+	outBuf := make([]byte, length)
+	n, err := hkdf.Expand(hash, secret, label).Read(outBuf)
+	if err != nil {
+		return nil, fmt.Errorf("hkdf.expand.Read failed with error: %v", err)
```

Comment:
Sorry I forgot to mention this in my last round of comments. Could we also add a test case to hit this return statement?

Reply:
Done